### PR TITLE
Action for update submodule and create PR

### PR DESCRIPTION
**Pull request by foxzi**
New action which can be used for update parent repository after updating current repository.
If You use sub-modules and want update parent repository after updating repository You can use this action. This is action create PR in parent repository after updating.

### CODE REVIEW

**Review by artembo — APPROVED** [review]

File: update-submodule/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  build:
+    name: Submodule update
+    runs-on: docker
+    env:
+      PARENT_REPOSITORY: 'org/repo'
+      CHECKOUT_BRANCH: 'main'
+      PR_AGAINST_BRANCH: 'main'
+      PARENT_UPDATED_BRANCH: 'branch-for-update-'
```
Comment (NickVolynkin) [inline update-submodule/README.md:26]:
```suggestion
      PARENT_UPDATED_BRANCH: 'update-submodule-name'
```

File: update-submodule/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      TEXT_PULL_REQUEST: 'Submodule updated time for renew'
```
Comment (NickVolynkin) [inline update-submodule/README.md:27]:
```suggestion
      PULL_REQUEST_TEXT: 'Update submodule <name> on branch <checkout branch>'
```

File: update-submodule/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# GitHub Action: Creates Pull Request when Submodules are Updated
+
+This GitHub action creates a new branch and pull request against the parent repository when submodules are updated.
```
Comment (NickVolynkin) [inline update-submodule/README.md:3]:
```suggestion
This action updates a submodule of the parent repository in
a feature branch and opens a pull request against the main branch.
```

File: update-submodule/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+```yml
+
```
Comment (NickVolynkin) [inline update-submodule/README.md:7]:
```suggestion
```

File: update-submodule/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: run action
+        id: run_action
+        uses: tarantool/action/update-submodule@master
+        with:
+          github_token: ${{ secrets.RELEASE_HUB_SECRET }}
```
Comment (NickVolynkin) [inline update-submodule/README.md:39]: what's that?

Comment (foxzi) [reply update-submodule/README.md:39]: It is token with permission to push branch and create pr.

File: update-submodule/README.md (hunk 1 above)
Comment (foxzi) [reply update-submodule/README.md:26]: It is only example. Doesn't matter how shown in readme.

File: update-submodule/README.md (hunk 4 above)
Comment (foxzi) [reply update-submodule/README.md:14]: ok, done

File: update-submodule/README.md (hunk 5 above)
Comment (foxzi) [reply update-submodule/README.md]: I added explanation

File: update-submodule/action.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      repository: ${{ inputs.parent_repository }}
+      ref: ${{ inputs.checkout_branch }}
+      submodules: true
+      fetch-depth: 0
+
+  - name: Create new branch and push changes
+    shell: bash
+    run: |
+      git config user.name github-actions
+      git config user.email github-actions@github.com
```
[the remark this answers is not in the document]
Comment (foxzi) [reply update-submodule/action.yml:56]: ok

File: update-submodule/README.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+This action updates a submodule of the parent repository in a feature branch and opens a pull request against the main branch.
+
+## Notice
+ - This action doesn't create another PR for all merges, commits for on branch, all updates in one PR
+ - Name of commit and title and body of PR have same text and can be set by variable TEXT_PULL_REQUEST
+ - variable owner can be set as organization
+ - branch name for updating can be set in variable PARENT_BRANCH_FOR_BRANCH
+ - parrent repoitory can be set by secret - ***PARENT_REPOSITORY: '${{ secret.PARENT_REPO }}***
+ - variable PARENT_REPO_TOKEN is token with permisson to
```
Comment (NickVolynkin) [inline update-submodule/README.md:11]:
```suggestion
## Variables

- `PARENT_REPOSITORY` 
- `OWNER`
- `CHECKOUT_BRANCH`
- `PR_AGAINST_BRANCH`
- `PARENT_BRANCH_FOR_UPDATE`
- `PULL_REQUEST_TEXT`

## Usage

- If there is an open PR already, the action will just update and force-push the branch.
- Commit message, PR title, and PR body have the same text, provided in the `PULL_REQUEST_TEXT` variable.
- Variable `OWNER` can 
 - This action doesn't create another PR for all merges, commits for on branch, all updates in one PR.
 - Name of commit and title and body of PR have same text and can be set by variable TEXT_PULL_REQUEST
 - variable owner can be set as organization
 - branch name for updating can be set in variable PARENT_BRANCH_FOR_BRANCH
 - parrent repoitory can be set by secret - ***PARENT_REPOSITORY: '${{ secret.PARENT_REPO }}***
 - variable PARENT_REPO_TOKEN is token with permisson to
```

File: update-submodule/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+          github_token: ${{ secrets.PARENT_REPO_TOKEN }}
+          parent_repository: ${{ env.PARENT_REPOSITORY }}
+          checkout_branch: ${{ env.CHECKOUT_BRANCH }}
+          pr_against_branch: ${{ env.PR_AGAINST_BRANCH }}
+          parent_branch_for_update: ${{ env.PARENT_BRANCH_FOR_UPDATE }}
+          pull_request_text: ${{ env.PULL_REQUEST_TEXT }}
+          owner: ${{ env.OWNER }}
+          branch: ${{ github.head_ref || github.ref_name }}
+          current_repo: "${{ github.repository }}"
+
```
Comment (ylobankov) [inline update-submodule/README.md:54]: Please remove this empty line.

**Review by ylobankov — COMMENTED** [review]
Will be all commits squashed before merge? If no, please fix the title of the first commit.

File: update-submodule/README.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (NickVolynkin) [inline update-submodule/README.md:58]:
```suggestion
```

**Review by NickVolynkin — APPROVED** [review]

**Pull request merged**